src/test/ui/suggestions/suggest-move-types.stderr

```diff
@@ -1,81 +1,69 @@
-  --> $DIR/suggest-move-types.rs:28:20
+error: generic arguments must come before the first constraint
+  --> $DIR/suggest-move-types.rs:26:26
    |
 LL | struct A<T, M: One<A=(), T>> {
-   |                    ----^^^
+   |                    ----  ^
    |                    |
-   |                    this associated type binding should be moved after the generic parameters
+   |                    the first constraint is provided here
 
-error: associated type bindings must be declared after generic parameters
-  --> $DIR/suggest-move-types.rs:34:37
+error: generic arguments must come before the first constraint
+  --> $DIR/suggest-move-types.rs:33:43
    |
 LL | struct Al<'a, T, M: OneWithLifetime<A=(), T, 'a>> {
-   |                                     ----^^^^^^^
+   |                                     ----  ^  ^^
    |                                     |
-   |                                     this associated type binding should be moved after the generic parameters
+   |                                     the first constraint is provided here
 
-error: associated type bindings must be declared after generic parameters
-  --> $DIR/suggest-move-types.rs:41:28
+error: generic arguments must come before the first constraint
+  --> $DIR/suggest-move-types.rs:40:46
    |
 LL | struct B<T, U, V, M: Three<A=(), B=(), C=(), T, U, V>> {
-   |                            ----^^----^^----^^^^^^^^^
-   |                            |     |     |
-   |                            |     |     this associated type binding should be moved after the generic parameters
-   |                            |     this associated type binding should be moved after the generic parameters
-   |                            this associated type binding should be moved after the generic parameters
+   |                            ----              ^  ^  ^
+   |                            |
+   |                            the first constraint is provided here
 
-error: associated type bindings must be declared after generic parameters
-  --> $DIR/suggest-move-types.rs:48:53
+error: generic arguments must come before the first constraint
+  --> $DIR/suggest-move-types.rs:48:71
    |
 LL | struct Bl<'a, 'b, 'c, T, U, V, M: ThreeWithLifetime<A=(), B=(), C=(), T, U, V, 'a, 'b, 'c>> {
-   |                                                     ----^^----^^----^^^^^^^^^^^^^^^^^^^^^
-   |                                                     |     |     |
-   |                                                     |     |     this associated type binding should be moved after the generic parameters
-   |                                                     |     this associated type binding should be moved after the generic parameters
-   |                                                     this associated type binding should be moved after the generic parameters
+   |                                                     ----              ^  ^  ^  ^^  ^^  ^^
```

Comment:
Do we not still want the label here?

Reply:
I thought you wanted to avoid the long label repeated on each generic argument. I'll add a shorter one.

Reply:
Added a shorter label now.

Reply:
Sorry, I meant I thought we should have one long label (the original one), but the spans it should be associated with are all the spans of the generic parameters that need to be reordered. That way we don't have many duplicate lines, but we still point all the generic parameters out, with a helpful message.

Reply:
I don't think that works out visually. If we have `A, B = u8, C,  D = u8, E`, then you would get a span underlining `D` as well, and that wouldn't really visually separate `D` from the pack.

Also, I don't suspect that the common case will have more than 1 constraint and one generic argument, so I think the diagnostic as is optimizes well for the common case.

Reply:
cc @estebank: this seems like a good place to be able to use `MultiSpan` with `span_label`.

Reply:
For these kind of cases I usually end up giving a label to the last of the spans in a `MultiSpan`, but we could encode that for `span_label<S: Into<MultiSpan>>(&self, span: S, label: &str)` in the API.

Reply:
I would probably add a label along these lines

```
error: generic arguments must come before the first constraint
  --> $DIR/suggest-move-types.rs:48:71
   |
LL | struct Bl<'a, 'b, 'c, T, U, V, M: ThreeWithLifetime<A=(), B=(), C=(), T, U, V, 'a, 'b, 'c>> {
   |                                                     ----              ^  ^  ^  ^^  ^^  ^^ the constraint must come after these generic parameters
   |                                                     |
   |                                                     the first constraint is provided here
```

Also, can we not catch all of `A`, `B` and `C`, like we used to?

Reply:
@estebank: this is exactly the sort of thing I was thinking :+1:

> Also, can we not catch all of A, B and C, like we used to?

I suppose we can, but @Centril thought that it's only the first one that's really relevant for repositioning.